pkg/config/config-map_test.go

```diff
@@ -64,11 +64,33 @@ var _ = Describe("ConfigMap", func() {
 				},
 			},
 		})
+		vmi.Spec.Domain.Devices.Disks = append(vmi.Spec.Domain.Devices.Disks, v1.Disk{
+			Name: "configmap-volume",
+		})
 
 		err := CreateConfigMapDisks(vmi, false)
 		Expect(err).NotTo(HaveOccurred())
 		_, err = os.Stat(filepath.Join(ConfigMapDisksDir, "configmap-volume.iso"))
 		Expect(err).NotTo(HaveOccurred())
 	})
 
+	It("Should not create a new config map iso disk without a Disk device", func() {
+		vmi := api.NewMinimalVMI("fake-vmi")
+		vmi.Spec.Volumes = append(vmi.Spec.Volumes, v1.Volume{
+			Name: "configmap-volume",
+			VolumeSource: v1.VolumeSource{
+				ConfigMap: &v1.ConfigMapVolumeSource{
+					LocalObjectReference: k8sv1.LocalObjectReference{
+						Name: "test-config",
+					},
+				},
+			},
+		})
+
+		err := CreateConfigMapDisks(vmi, false)
+		Expect(err).NotTo(HaveOccurred())
+		_, err = os.Stat(filepath.Join(ConfigMapDisksDir, "configmap-volume.iso"))
+		Expect(err).To(HaveOccurred())
```

Comment:
Probably better to be more explicit here:
```suggestion
		Expect(err).To(MatchError(os.ErrNotExist))
```
Same with other tests.

Reply:
Or maybe we should simply check that the directory is empty or has the config map file inside? I know that we are switching from having iso to files to directly using the files, but for the test itself I think it should test the new behavior in general and not verify that the old one still applies.

Reply:
> Or maybe we should simply check that the directory is empty or has the config map file inside? I know that we are switching from having iso to files to directly using the files, but for the test itself I think it should test the new behavior in general and not verify that the old one still applies.

I'm not sure to understand, we're keeping the old behavior of generating iso files, just with the restriction of creating them only if a Disk device exists. Or you are suggesting that we should also check if the ConfigMap volume was created correctly?

Reply:
It was not very clear, let me rephrase.
My comment was more about the purpose of the test. This test now checks that the old behavior doesn't apply, in other words, checking that the iso doesn't exist in the path. 
I understand the reasoning behind it, but I think it would make more sense if we check in the test, that the new behavior applies instead of checking that the old one doesn'appöy anymore. For example, if you expect an empty directory, then check that the directory is empty. 
I hope I manage to express it in a better way, let me know

Reply:
Yes, thanks for the clarification. I'll change the tests.